test/validate/validate.go

```diff
@@ -224,3 +224,94 @@ func (v *Validator) ValidateDualStackControlPlane(ctx context.Context) error {
 
 	return nil
 }
+
+func (v *Validator) validateIPv4NodeProperties(ctx context.Context) error {
```

Comment:
This function is similar to `validateDualStackNodeProperties`, i would say we create function in validator i.e `validateNodeProperties` and reuse the code if we can

Reply:
Good point, @paulyufan2 , you could make a function like this:
```
func validateNodeProperties(labels map[string]string, ipCount int) {
nodes, err := k8sutils.GetNodeListByLabelSelector(ctx, v.clientset, nodeSelectorMap[v.os])
	if err != nil {
		return errors.Wrap(err, "failed to get node list")
	}

	for index := range nodes.Items {
		nodeName := nodes.Items[index].ObjectMeta.Name
		// check nodes status;
		// nodes status should be ready after cluster is created
		nodeConditions := nodes.Items[index].Status.Conditions
		if nodeConditions[len(nodeConditions)-1].Type != corev1.NodeReady {
			return errors.Wrapf(err, "node %s status is not ready", nodeName)
		}

		// get node labels
		nodeLabels := nodes.Items[index].ObjectMeta.GetLabels()
		for key := range nodeLabels {
			if label, ok := labels[key]; ok {
				log.Printf("label %s is correctly shown on the node %+v", key, nodeName)
				if label != overlayClusterLabelName {
					return errors.Wrapf(err, "node %s overlay label name is wrong; expected label:%s but actual label:%s", nodeName, overlayClusterLabelName, label)
				}
			}
		}

		// check if node has two internal IPs(one is IPv4 and another is IPv6)
		internalIPCount := 0
		for _, address := range nodes.Items[index].Status.Addresses {
			if address.Type == "InternalIP" {
				internalIPCount++
			}
		}
		if internalIPCount != ipCount {
			return errors.Wrap(err, "node does not expected number of IPs")
		}
}
```

Reply:
And then call it like this:
```
func (v *Validator) validateDualStackNodeProperties(ctx context.Context) error {
    log.Print("Validating Dualstack Overlay Node properties")
    if err := validateNodeProperties(ctx, dualstackOverlayNodeLabels, 2); err != nil  {
        return err
    }
return nil
```
func (v *Validator) validateIPv4NodeProperties(ctx context.Context) error {
    log.Print("Validating IPv6 Overlay Cluster Node properties")
    if err := validateNodeProperties(ctx, v4OverlayNodeLabels, 1); err != nil  {
        return err
    }
return nil
```

```